packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -890,9 +892,11 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
       // Initial render
       changedBits = MAX_SIGNED_31_BIT_INT;
     } else {
+      const canBailOnChildren =
```

Comment:
Hoisting this out isn't free. Property access can be expensive in JS so it matter when we do it.

What is worse is that it can eat registers so we should avoid local variables if they're not in an immediate local place where they can be inlined.

Reply:
OK, I inlined it back at usage.